stdlib/ballerina-http/src/main/ballerina/http/annotation.bal

```diff
@@ -38,6 +40,18 @@ public type HttpServiceConfig {
     CorsConfig cors,
     Versioning versioning,
     ListenerAuthConfig? authConfig,
+    PayloadOverflowConfig overflowConfig,
+};
+
+documentation {
+    Configuration to control payload location(memory/temp file) based on a given threshold value.
```

Comment:
Shall we be a bit more descriptive about this feature here? For example, what this is, why and when is it needed, how this behaves (e.g., what does Ballerina do with the saved payloads, what happens to them when they are not needed anymore)